bindings/python/pymongoarrow/api.py

```diff
@@ -316,6 +318,21 @@ def _tabular_generator(tabular):
             return
 
 
+class _PandasNACodec(TypeCodec):
+    """A custom type codec for Pandas NA objects."""
+
+    python_type = NA.__class__  # type:ignore[assignment]
+    bson_type = None  # type:ignore[assignment]
```

Comment:
Interesting, why the type ignores?

Reply:
My editor had flagged them because the base class uses read-only properties.

Reply:
I see mypy does not raise this issue. This does look like a bug in pyright though:
```
  /Users/shane/git/mongo-python-driver/ignore.py:15:19 - error: Expression of type "python_type" cannot be assigned to declared type "property"
    "Type[type]" is incompatible with "Type[property]" (reportGeneralTypeIssues)
```

Looks like https://github.com/microsoft/pyright/issues/2601 which they closed as won't fix.

Reply:
Actually that pyright issue has `@classmethod` whereas we just use `@property` so I think it would be worth opening a new pyright issue for it.

Reply:
https://github.com/microsoft/pyright/issues/4364